packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -396,6 +419,24 @@ public void isSignedIn(final Result result) {
       result.success(value);
     }
 
+    @Override
+    public void recoverAuth(Result result, String accountId) {
+      if (currentAccount == null
+          || userRecoverableAuthIntent == null
+          || !accountId.equals(currentAccount.getId())) {
```

Comment:
I don't think this will work for us. We have use cases where we try to get the token when no account is selected. We try to get the token for every account before any account is selected. 

Couple ideas:

- Store every Intent. While this leaks memory, I think intents are pretty tiny and don't hold any resources.

- Map of account to Intent. Slightly better than above.

- Have an *option* in the get token method to automatically launch the recovery flow if necessary.

Reply:
cc @jacobklintg who thought about this for iOS

Reply:
Alan covered the cases pretty well. We also want to be able to support silent sign-in attempts, which explicitly do not show any UI.

I think an optional boolean parameter at the dart level (defaulting to true) to control whether or not to attempt reauth when needed will work for us on both platforms.

It might be simplest on the app developer if the plugin just passes this boolean along to the native code and lets it perform reauth if desired, without making the developer explicitly perform reauth. Either the boolean is true so the developer wants reauth (which I imagine is the common case), or else it's false and the call will fail if reauth is needed.

Reply:
@alanrussian @jklint-g I added an option to attempt to recover auth automatically once. If it doesn't work, authentication will return null.